static/tests/frontend/specs/scenesLength.js

```diff
@@ -0,0 +1,134 @@
+    });
+  }
+
+  var resetEventData = function() {
+    scriptLengthChangedEvent = false;
+    scriptLengthEventData = {};
+  }
+
+  before(function(done) {
+    var lastElement = 'last heading';
+    utils.newPad(function(){
+      var lastSceneTitle = 'last scene';
+      var createScene = function(text) { return utils.synopsis(text) + utils.heading(text)}
+      var script = createScene('scene') + createScene(lastSceneTitle);
+      utils.createScriptWith(script, lastSceneTitle, function() {
+        listenToEventScriptLengthChanged();
+        waitToBuildScenesLengthObj(done);
+      });
+    });
+    this.timeout(20000);
+  });
+
+  it('saves the scenes length', function(done) {
+    var scenesLength = getScenesLength();
+    var lineDefaultSize = getLineDefaultSize();
+    expect(scenesLength[0]).to.be(lineDefaultSize * 3);
+    expect(scenesLength[1]).to.be(lineDefaultSize * 3);
+    done();
+  });
+
+  context('when an edition does not change the scene length', function() {
+    var originalSceneLength;
+    before(function(done) {
+      resetEventData();
+      originalSceneLength = getScenesLength();
+      var $lastHeading = helper.padInner$('heading').last();
+      $lastHeading.sendkeys('{selectall}{rightarrow}EDITED');
+      setTimeout(function() {
+        done();
+      }, 4000)
+      this.timeout(6000);
+    });
+
+    after(function() {
+      utils.undo();
+    });
+
+    it('does not change the scenes length object', function(done) {
+      var scenesLength = getScenesLength();
+      expect(originalSceneLength[0]).to.be(scenesLength[0]);
+      expect(originalSceneLength[1]).to.be(scenesLength[1]);
+      expect(originalSceneLength.length).to.be(scenesLength.length);
+      done();
+    });
+
+    it('does not trigger the scenes length change event', function(done) {
+      expect(scriptLengthChangedEvent).to.be(false);
+      expect(scriptLengthEventData).to.be.empty();
+      done();
+    });
+  });
+
+  context('when scene length changes', function() {
+    before(function(done) {
+      resetEventData();
+      utils.changeToElement(utils.GENERAL, function() {
+        helper.waitFor(function() {
+          // as we've removed the second scene so we wait to update the scenes
+          // length object to only 1 element
+          return scriptLengthChangedEvent;
+        }, 4000).done(done);
+      }, LAST_HEADING_LINE);
+      this.timeout(6000);
+    });
+
+    after(function() {
+      utils.undo();
+    });
+
+    it('updates the scenes length object', function(done) {
+      var scenesLength = getScenesLength();
+      var expectedSceneLength = getLineDefaultSize() * 4; // 1 heading + 1 general = 4 lines
+      expect(scenesLength[0]).to.be.within(
+        expectedSceneLength - TOLERANCE,
+        expectedSceneLength + TOLERANCE
+      );
+      expect(scenesLength.length).to.be(1);
+      done();
+    });
+
+    context('ep_scene_navigator - integration with navigator data', function() {
```

Comment:
na verdade esse teste deveria estar no `ep_scene_navigator`, não? (Com o prefixo do `ep_SE`)

Reply:
fiquei na dúvida por que aqui a gente testa se manda um evento com uma flag no data. Como é esse plugin que dispara isso, não seria mais correto deixar aqui? Acho que dentro do `ep_scene` deveria testar o que faz quando recebe um evento com essa flag, não?

Reply:
Depois que revisei o `ep_navigator` vi que tem um teste lá que vai quebrar se o evento não for enviado. 

O que eu tinha pensado quando te falei de "validar o evento nos testes colocando o nome do plugin" foi mencionar o `ep_SE` no nome do teste do navigator mesmo, para que aquele teste fosse rodado quando a gente mudasse o `ep_SE`. Mas testar as duas pontas não é problema não, na verdade deixa até mais fácil de ver que aqui quebrou algo. Só não faz sentido eu rodar esse teste aqui quando eu mudar o `ep_navigator` -- não tem nada lá que vá quebrar algo daqui. Por isso que eu falei de inverter onde vai o teste.

Ficou mais claro o que eu quis dizer?

Reply:
entendi teu ponto